gph/src/ripser.h

```diff
@@ -706,11 +729,19 @@ class ripser
                                         index_t n, OutputIterator out) const
     {
         --n;
-        for (index_t k = dim + 1; k > 0; --k) {
+        for (index_t k = dim + 1; k > 2; --k) {
             n = get_max_vertex(idx, k, n);
             *out++ = n;
             idx -= binomial_coeff(n, k);
         }
+
+        double to_sqrt = 8 * idx + 1;
+        n = static_cast<index_t>(std::floor(1.5 + (0.5 * std::sqrt(to_sqrt)))) - 1;
+        *out++ = n;
+        idx -= binomial_coeff(n, 2);
+        *out++ = idx;
+        idx = 0;
```

Comment:
This should be simplified to:

```cpp
*out = idx;
```

No need the increment and the `idx=0`

Reply:
The `idx=0` is just there to have consistent behaviour with the previous version, which always ended there. Mostly for debugging now, we can remove it at the end!

Reply:
This should provide a measurable speed-up IMO

Reply:
We have to make sure `idx` is not used ever after calling this, I think there is an instance in the code where that is the case,

Reply:
I am not sure I agree. After the last `*out=idx` we do not need anymore `idx`. Inside of function `get_simplex_vertices` the `idx` value was passed by copy, it is not a reference. It will be discarded after we arrive at the end of the function.

Reply:
> Inside of function get_simplex_vertices the idx value was passed by copy, it is not a reference.

Thanks for the clarification!  Yes then you are 100% right.

Reply:
Done.